modules/reactor/src/meshgenerators/CoarseMeshExtraElementIDGenerator.C

```diff
@@ -53,11 +57,32 @@ CoarseMeshExtraElementIDGenerator::generate()
   std::unique_ptr<MeshBase> mesh = std::move(_input);
 
   unsigned int coarse_id;
-  if (!mesh->has_elem_integer(_coarse_id_name))
+  const bool already_has_id = mesh->has_elem_integer(_coarse_id_name);
+  if (!already_has_id)
     coarse_id = mesh->add_elem_integer(_coarse_id_name);
   else
     coarse_id = mesh->get_elem_integer_index(_coarse_id_name);
 
+  // If we are not going through the entire mesh and the ID already exists, we need to offset the
+  // assigned ID so that we don't copy ones that are already there
+  dof_id_type id_offset = 0;
+  if (already_has_id && isParamValid("subdomains"))
```

Comment:
`isParamValid("subdomains")` here will always return true, unless remove its default value in the validParams. In that case, Line 83 will need to be protected.